### PR TITLE
Introducing swipeToClosePercent

### PR DESCRIPTION
**Pull request by noitcudni**
Also, expose swipeGestuerBegan callback.

openCellId wasn't being reset when a row is closed.

Shipping down rowData for onRowOpen, onRowDidOpen, onRowClose, and onRowDidClose to increase flexibility.

### CODE REVIEW

File: components/SwipeListView.js [hunk 1]
```diff
@@ -38,15 +38,26 @@ class SwipeListView extends Component {
 		if (this.props.closeOnRowBeginSwipe && this.openCellId && this.openCellId !== id) {
 			this.safeCloseOpenRow();
 		}
+
+    if (this.props.swipeGestureBegan) {
```
Comment (jemise111) [inline components/SwipeListView.js]: Looks like the formatting got messed up here and everywhere else. Could you make sure you're using tabbed indentation and just see if the formatting matches the rest of the file, thanks!

File: components/SwipeListView.js [hunk 2]
```diff
@@ -38,15 +38,26 @@ class SwipeListView extends Component {
...
 
-	onRowOpen(secId, rowId, rowMap) {
+	onRowOpen(rowData, secId, rowId, rowMap) {
 		const cellIdentifier = `${secId}${rowId}`;
 		if (this.openCellId && this.openCellId !== cellIdentifier) {
 			this.safeCloseOpenRow();
 		}
 		this.openCellId = cellIdentifier;
-		this.props.onRowOpen && this.props.onRowOpen(secId, rowId, rowMap);
+		this.props.onRowOpen && this.props.onRowOpen(rowData, secId, rowId, rowMap);
```
Comment (jemise111) [inline components/SwipeListView.js]: Hm, introducing `rowData` here and into all of the other callbacks as the first argument is going to be a big breaking change that I don't feel is necessary. If anything it should be the 4th argument.

But do we need to pass `rowData` at all? You already have the `rowMap` so you can access any row you need. And the parent component has to pass `rowData` to the `SwipeListView` so why do we need to pass it back to the parent in the callback?

I'm not sure I see the use case here

File: components/SwipeRow.js [hunk 3]
```diff
@@ -144,17 +144,38 @@ class SwipeRow extends Component {
...
-				toValue = this.props.leftOpenValue;
-			}
+			// trying to swipe right
+
+      if (this.swipeInitialX < this._translateX._value) {
+        if (this._translateX._value > this.props.leftOpenValue * (this.props.swipeToOpenPercent/100)) {
+          // we're more than halfway
+          toValue = this.props.leftOpenValue;
+        } else {
+          toValue = 0;
```
Comment (jemise111) [inline components/SwipeRow.js]: No need to set `toValue` back to 0 here or below, it's the default value (see line 145)

**Comment by noitcudni** [comment]
Hey, @jemise111. I'll take a look some time this week. Thanks for looking it over.

File: components/SwipeListView.js (hunk 2 above)
Comment (noitcudni) [reply components/SwipeListView.js]: To be honest, I don't remember why I had it. I do find it annoying that I don't have access to the data being passed to a row object.  Anyway, I'm OK to get rid of this.

**Comment by noitcudni** [comment]
@jemise111 
Resolved conflicts around `swipeToOpenVelocityContribution`.  However, you might want to double check, especially in `SwipeRow.js` line 169, 174, 181, and 186.

Made changes/reformatting per your comments.

File: components/SwipeListView.js [hunk 4]
```diff
@@ -39,6 +39,11 @@ class SwipeListView extends Component {
 		if (this.props.closeOnRowBeginSwipe && this.openCellId && this.openCellId !== id) {
 			this.safeCloseOpenRow();
 		}
+
+		if (this.props.swipeGestureBegan) {
```
Comment (jemise111) [inline components/SwipeListView.js]: Would be awesome if you could throw this into `PropTypes` and also throw a note in the README

File: components/SwipeListView.js [hunk 5]
```diff
@@ -50,6 +55,12 @@ class SwipeListView extends Component {
 		this.props.onRowOpen && this.props.onRowOpen(secId, rowId, rowMap);
 	}
 
+	onRowClose(secId, rowId, rowMap) {
```
Comment (jemise111) [inline components/SwipeListView.js]: I can't see where this is being used, unless I'm missing something? If that's the case can you remove this function please

File: components/SwipeRow.js [hunk 6]
```diff
@@ -412,6 +424,11 @@ SwipeRow.propTypes = {
 	 * and it'll just take into consideration the swipeToOpenPercent.
 	 */
 	swipeToOpenVelocityContribution: PropTypes.number,
+  /**
```
Comment (jemise111) [inline components/SwipeRow.js]: sorry for the nitpicking, just fix the indentation here please

File: components/SwipeListView.js (hunk 4 above)
Comment (noitcudni) [reply components/SwipeListView.js]: Done

**Comment by noitcudni** [comment]
Added swipeGestureBegan to README and PropTypes.

**Pull request merged**